### PR TITLE
docs: update docs for collect list/set

### PR DESCRIPTION
**Pull request by agavra**
### Description 

Now that #6711 is closed, we should update the docs accordingly!

### Testing done 
n/a

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/aggregate-functions.md [hunk 1]
```diff
@@ -31,12 +31,14 @@ Stream, Table
 Return an array containing all the values of `col1` from each
 input row (for the specified grouping and time window, if any).
 Currently only works for simple types (not Map, Array, or Struct).
-This version limits the size of the result Array to a maximum of
-1000 entries and any values beyond this limit are silently ignored.
+
+The size of the result Array can be limited to a maximum of
+`ksql.functions.collect_list.limit` entries and any values beyond this
```
Comment (MichaelDrogalis) [inline docs/developer-guide/ksqldb-reference/aggregate-functions.md:36]: I think these params also need to be documented in https://docs.ksqldb.io/en/latest/reference/server-configuration/

**Review by MichaelDrogalis — CHANGES_REQUESTED** [review]

File: docs/developer-guide/ksqldb-reference/aggregate-functions.md [hunk 2]
```diff
@@ -31,12 +31,14 @@ Stream, Table
...
-1000 entries and any values beyond this limit are silently ignored.
+
+The size of the result Array can be limited to a maximum of
+`ksql.functions.collect_list.limit` entries and any values beyond this 
+limit are silently ignored.
 When using with a window type of `session`, it can sometimes
 happen that two session windows get merged together into one when a
 out-of-order record with a timestamp between the two windows is
-processed. In this case the 1000 record limit is calculated by
+processed. In this case the record limit is calculated by
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/aggregate-functions.md:41]:
```suggestion
processed. In this case, the record limit is calculated by
```

File: docs/developer-guide/ksqldb-reference/aggregate-functions.md [hunk 3]
```diff
@@ -54,12 +56,14 @@ Stream
...
-1000 entries and any values beyond this limit are silently ignored.
+
+The size of the result Array can be limited to a maximum of
+`ksql.functions.collect_set.limit` entries and any values beyond this
+limit are silently ignored.
 When using with a window type of `session`, it can sometimes
 happen that two session windows get merged together into one when a
 out-of-order record with a timestamp between the two windows is
-processed. In this case the 1000 record limit is calculated by
+processed. In this case the record limit is calculated by
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/aggregate-functions.md:66]:
```suggestion
processed. In this case, the record limit is calculated by
```

File: docs/reference/server-configuration.md [hunk 4]
```diff
@@ -101,6 +101,20 @@ to your ksqlDB Server properties file:
 ksql.fail.on.production.error=false
 ```
 
+## `ksql.functions.collect_list.limit`
+
+Allows users to limit the size of the resultant Array to N entries, beyond which
```
Comment (JimGalasyn) [inline docs/reference/server-configuration.md:106]:
```suggestion
Limit the size of the resultant Array to N entries, beyond which
```

File: docs/reference/server-configuration.md [hunk 5]
```diff
@@ -101,6 +101,20 @@ to your ksqlDB Server properties file:
...
+any further values will be silently ignored, by setting this configuration to N.
```
Comment (JimGalasyn) [inline docs/reference/server-configuration.md:107]:
```suggestion
any further values are silently ignored, by setting this configuration to N.
```

File: docs/reference/server-configuration.md [hunk 6]
```diff
@@ -101,6 +101,20 @@ to your ksqlDB Server properties file:
...
+
+Also see [aggregate-functions](/reference/aggregate-functions)
+
+## `ksql.functions.collect_set.limit`
+
+Allows users to limit the size of the resultant Set to N entries, beyond which
```
Comment (JimGalasyn) [inline docs/reference/server-configuration.md:113]:
```suggestion
Limits the size of the resultant Set to N entries, beyond which
```

File: docs/reference/server-configuration.md [hunk 7]
```diff
@@ -101,6 +101,20 @@ to your ksqlDB Server properties file:
...
+any further values will be silently ignored, by setting this configuration to N.
```
Comment (JimGalasyn) [inline docs/reference/server-configuration.md:114]:
```suggestion
any further values are silently ignored, by setting this configuration to N.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a few suggestions.

**Review by MichaelDrogalis — APPROVED** [review]

**Pull request merged**